spring/src/main/java/org/axonframework/spring/config/AxonConfiguration.java

```diff
@@ -78,12 +80,24 @@ public EventBus eventBus() {
         return config.eventBus();
     }
 
+    @Override
+    public QueryUpdateEmitter queryUpdateEmitter() {
+        return config.queryUpdateEmitter();
+    }
+
     @NoBeanOfType(QueryBus.class)
     @Bean("queryBus")
+    @Primary
```

Comment:
Curious: why did you need the `@Primary` annotation here?

Reply:
By default, queryUpdateEmitter is also a QueryBus. So, we'll have to query buses in application context, one has to be primary.

Reply:
Ah, I actually thought we'd just have one bean which would fulfil both roles! 
Isn't creating two identical beans, were each bean only does halve the work it's capable off, kinda weird?

Reply:
It's the same thing again: whether to separate `QueryBus` and `QueryUpdateEmitter` or not? The thing you explained here can be achieved if `QueryBus` extends `QueryUpdateEmitter`. So far, I think that we'are heading towards this approach. Would like to hear @abuijze 's opinion on this.

Reply:
Like you suggested a bit earlier in this PR, I'd say we can postpone that decision until we've got another concrete implementation of a `QueryBus`. For now, this'll work I assume, so let's go with it.